src/app/Fake.Core.Target/Target.fs

```diff
@@ -318,9 +322,9 @@ module Target =
           |> Seq.map (fun kv -> kv.Key)
           |> Seq.fold (fun context name ->
                         let target = get name
-                        use t = Trace.traceFinalTarget target.Name (match target.Description with Some d -> d | _ -> null) (dependencyString target)
+                        use t = Trace.traceFinalTarget target.Name target.Description (dependencyString target)
                         let res = runSimpleContextInternal target context
-                        if res.HasError
+                        if res.LastTargetError
```

Comment:
Another way would be to let runSimpleContextInternal return the state of the target it executed

Reply:
ooooh even nicer!